Rust/src/lib.rs

```diff
@@ -493,6 +497,24 @@ impl FastNoiseLite {
         -0.7870349638, 0.03447489231, 0.6159443543, 0., -0.2015596421, 0.6859872284, 0.6991389226, 0., -0.08581082512, -0.10920836, -0.9903080513, 0., 0.5532693395, 0.7325250401, -0.396610771, 0., -0.1842489331, -0.9777375055, -0.1004076743, 0., 0.0775473789, -0.9111505856, 0.4047110257, 0., 0.1399838409, 0.7601631212, -0.6344734459, 0., 0.4484419361, -0.845289248, 0.2904925424, 0.,
     ];
 
+    #[inline(always)]
+    fn fast_abs(f: f32) -> f32 {
+        if f < 0. {
+            -f
+        } else {
+            f
+        }
+    }
+
+    /// Accurate to within roughly 6% at worst.
+    ///
+    /// Note: Requires f >= 0.
+    #[inline(always)]
+    fn fast_sqrt(f: f32) -> f32 {
+        // Explanation: https://stackoverflow.com/questions/41785416/how-does-this-sqrt-approximation-inline-assembly-function-work
+        f32::from_bits((f.to_bits() + 0x3f80_0000) >> 1)
```

Comment:
Alternatively, we could port [this](https://github.com/simonbyrne/apple-libm/blob/master/Source/ARM/sqrt.c) for the accurate implementation?

We should maybe also consider switching between the std version and the no_std polyfill? But then the results would differ, so I'm not sure what to do exactly.

Reply:
Another option is to use this to refine the guess (it doesn't seem to iterate for more than several iterations, but it'll go until the numerical instability reaches insignificance and the guesses stop changing):

```rs
let mut guess = f32::from_bits((f.to_bits() + 0x3f80_0000) >> 1);

loop {
    let new_guess = 0.5 * (guess + f / guess);
    if new_guess == guess {
        break;
    }
    guess = new_guess;
}

guess
```